client/src/api/index.ts

```diff
@@ -1,47 +1,30 @@
-const url = "http://localhost:8000";
+import axios from "axios";
 
-export const apiGet = async (path:string, paramDic:any={}) => {
-  const init = {
-    method: 'GET',
-    headers: {
-      'Content-Type': 'application/json',
-      Accept: 'application/json',
-    },
-  };
 
-  const params = objToQueryString(paramDic);
-  const res = await fetch(`${url}${path}?${params}`, init);
-  // await console.log(res);
-  const body = await res.json();
-  if (body.code === 200) {
-    return body;
-  } else {
-    throw new Error(body.errmsg);
+export const apiGet = async (path: string, params: any = {}) => {
+  try {
+    let result = await axios.get(path, params);
+    if(result.status ===200) {
+      return result.data
+    }else {
+      throw new Error('data query failure')
+    }
+  } catch (error) {
+    alert(error)
+    // return error
```

Comment:
If you return 40x status codes when you get no data, you Axio will auto catch that error for you. so you don't need to test if `result.status===200` and throw and error. 

also as a convention, the client should not alert an error if no data is retrieved. simply not rendering them or log/throw something in the console should be OK.

You can simply your code to this:

```
try {
    let result = await axios.get(path, params);
    return result.data;
} catch (error) {
    console.log(error);
    throw new Error('data query failure');
```

Reply:
Alternatively your components should be able to handle empty [] or undefined returns.